tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -33,6 +35,84 @@ static PyObject * THPVariable_detach_(PyObject* self, PyObject* args)
   END_HANDLE_TH_ERRORS
 }
 
+static IntList dispatch_size(const Tensor& self) {
+  // avoid releasing GIL/changing device, should be quick
+  // yes, this is called sizes in ATen.
+  return self.sizes();
+}
+
+static int64_t dispatch_size(const Tensor& self, int64_t dim) {
+  // avoid releasing GIL/changing device, should be quick
+  return self.size(dim);
+}
+
+static PyObject * THPVariable_size(PyObject* self, PyObject* args, PyObject* kwargs)
+{
+  HANDLE_TH_ERRORS
+  static PythonArgParser parser({
+    "size(int64_t dim)",
+    "size()",
+  });
+  auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
+  PyObject* parsed_args[3];
+  auto r = parser.parse(args, kwargs, parsed_args);
+  if (r.idx == 0) {
+    return wrap(dispatch_size(self_, r.toInt64(0)));
+  } else if (r.idx == 1) {
+    // we can't do the normal wrapping here because IntList maps to both
+    // torch.Size and tuple in python
+    IntList sizes = dispatch_size(self_);
+    return THPSize_New(sizes.size(), (int64_t *)sizes.data());
+  }
+  Py_RETURN_NONE;
+  END_HANDLE_TH_ERRORS
+}
+
+static IntList dispatch_stride(const Tensor& self) {
+  // avoid releasing GIL/changing device, should be quick
+  // yes, this is called strides in ATen.
+  return self.strides();
+}
+
+static int64_t dispatch_stride(const Tensor& self, int64_t dim) {
+  // avoid releasing GIL/changing device, should be quick
+  return self.stride(dim);
+}
+
+static PyObject * THPVariable_stride(PyObject* self, PyObject* args, PyObject* kwargs)
+{
+  HANDLE_TH_ERRORS
+  static PythonArgParser parser({
+    "stride(int64_t dim)",
+    "stride()",
+  });
+  auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
+  PyObject* parsed_args[3];
+  auto r = parser.parse(args, kwargs, parsed_args);
+  if (r.idx == 0) {
+    return wrap(dispatch_stride(self_, r.toInt64(0)));
+  } else if (r.idx == 1) {
+    // we can't do the normal wrapping here because IntList maps to both
+    // torch.Size and tuple in python
+    IntList strides = dispatch_stride(self_);
+    THPObjectPtr py_stride(PyTuple_New(strides.size()));
+    for (size_t i = 0; i != strides.size(); ++i) {
+      PyTuple_SET_ITEM(py_stride.get(), i, PyLong_FromLong(strides[i]));
```

Comment:
Can you move this logic to it's own function? We do this in a bunch of places and we keep getting it wrong:

1. PyLong_FromLong returns null on error. We should check that.
2. We should prefer `THPUtils_packInt64` since `int` types are nicer on Python 2 when the values fit in an int.

You can make `THPSize_New` use this function too.